gorm-tools/src/main/groovy/gorm/tools/hibernate/AbstractQueryListener.groovy

```diff
@@ -0,0 +1,69 @@
+/*
+* Copyright 2024 Yak.Works - Licensed under the Apache License, Version 2.0 (the "License")
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+package gorm.tools.hibernate
+
+import groovy.transform.CompileDynamic
+import groovy.transform.CompileStatic
+
+import org.grails.orm.hibernate.query.AbstractHibernateQuery
+import org.grails.orm.hibernate.query.HibernateHqlQuery
+import org.hibernate.Criteria
+import org.hibernate.query.Query
+
+import yakworks.commons.lang.Validate
+
+import static yakworks.util.ReflectionUtils.getPrivateFieldValue
+
+@CompileStatic
+abstract class AbstractQueryListener {
+
+    //returns criteria or hql query based on type of query
```

Comment:
if your adding a comment to a method then its never a `//`, use proper javadoc format.